frontend/src/components/MapView/index.tsx

```diff
@@ -215,9 +215,26 @@ const MapView = memo(({ classes }: MapViewProps) => {
   const addMissingLayers = useCallback(() => {
     missingLayers.forEach(layerId => {
       const layer = LayerDefinitions[layerId as LayerKey];
+      const { serverLayerName } = layer as any;
+      if (
+        // The layer does not have date support
+        serverAvailableDates[serverLayerName] !== undefined &&
+        // The layer does have date support but no additional available dates are loaded
+        serverAvailableDates[serverLayerName].length === 0
```

Comment:
The top comment is a bit confusing imho.

Could this be simply: `serverAvailableDates[serverLayerName]?.length || -1) === 0`

Reply:
Yes the way I did this is because when a layer does not exist at all in the serverAvailableDates, which will have `undefined` value it means that the layer does not have date support but it can be enabled, only if the layer is included in the serverAvailableDates and has a length of 0 means that it does not have available dates.

Reply:
What you are actually checking is that the layer has date support BUT not dates were returned by the date endpoint.

Reply:
Can't you simply check `serverAvailableDates[serverLayerName]?.length === 0`

Reply:
Yes you are right I am changing it.